### PR TITLE
Fullpage SS

### PR DESCRIPTION
**Pull request by prklm10**

### CODE REVIEW

File: Percy/AppPercy.cs [hunk 1]
```diff
@@ -39,7 +39,7 @@ internal void setValues(IPercyAppiumDriver percyAppiumDriver)
       this.sessionId = percyAppiumDriver.sessionId();
     }
 
-    public void Screenshot(String name, ScreenshotOptions? options = null, Boolean fullScreen = false)
+    public void Screenshot(String name, ScreenshotOptions? options = null, Boolean fullScreen = false, String platformVersion = null,  int fullPageScreenshotScreenLengths = -1)
```
Comment (ninadbstack) [inline Percy/AppPercy.cs:42]: these params should be part of options

File: Percy/metadata/IosMetadata.cs [hunk 2]
```diff
@@ -86,5 +86,16 @@ internal override int StatBarHeight()
       }
       return (Dictionary<string, object>)AppPercy.cache.Get("viewportRect_" + sessionId);
     }
+
+    internal override int GetScaleFactor()
+    {
+      if (AppPercy.cache.Get("scaleFactor_" + sessionId) == null)
+      {
+        object scaleFactor;
+        driver.GetSessionDetails().TryGetValue("pixelRatio", out scaleFactor);
```
Comment (ninadbstack) [inline Percy/metadata/IosMetadata.cs:95]: this is already cached. so no need to cache this method.

File: Percy/providers/AppAutomate.cs [hunk 3]
```diff
@@ -106,8 +107,58 @@ internal JObject ExecutePercyScreenshotEnd(String name, String percyScreenshotUr
...
+      arguments.Add("options", args);
+
+      JObject reqObject = new JObject();
+      reqObject.Add("action", "percyScreenshot");
+      reqObject.Add("arguments", arguments);
+      Console.WriteLine(reqObject.ToString());
+      Console.Write(percyAppiumDriver.sessionId());
+      var resultString = percyAppiumDriver.ExecuteScript(string.Format("browserstack_executor: {0}", reqObject.ToString())).ToString();
+      JObject result = JObject.Parse(resultString);
+      return result.GetValue("result").ToString();
```
Comment (ninadbstack) [inline Percy/providers/AppAutomate.cs:157]: Maybe refactor this to a separate function. [ We must be using this in 3 locations by now ]

**Comment by pankaj443** [comment]
Closing since not taken care in other PR

**Pull request closed without merging**